### DISCUSSION TITLE
Is there a way to do a stream upload to firebase storage?

### DISCUSSION THREAD

**Discussion by bxclib2** [firebase/firebase-js-sdk#8539] [Q&A]
I want to stream the file chunk generated from the recorder. Like this:
```
                    recorder.ondataavailable = (event) => {
                        if (event.data.size > 0) {
                            streamToFirebase(event.data)
                            
                        }
                    };
                    recorder.onstop = event => {
                         streamToFirebase.close()
                    }
```
But the final file need to be a single file. Is there anyway to do this?

Thanks a lot!

**Comment by prathmesh235** [comment]
For Firebase Storage, you can use uploadBytesResumable() for a resumable upload, but it expects the data for the object being uploaded rather than treating each Blob chunk from MediaRecorder as an append operation.

If you want the final recording to be one file, a common approach is to collect the MediaRecorder chunks and create a single Blob when recording stops:

const chunks = [];

recorder.ondataavailable = (event) => {
  if (event.data.size > 0) {
    chunks.push(event.data);
  }
};

recorder.onstop = async () => {
  const blob = new Blob(chunks, { type: recorder.mimeType });
  const storageRef = ref(storage, `recordings/${Date.now()}.webm`);

  await uploadBytesResumable(storageRef, blob);
};

If you need the chunks to be uploaded while recording is still in progress, you would need a different architecture (for example, a backend/service that can append or assemble chunks), rather than calling Firebase Storage upload for every chunk as if it were an append operation.
